repo/blob/filesystem/osinterface_realos.go

```diff
@@ -58,6 +59,14 @@ func (realOS) Stat(fname string) (os.FileInfo, error) {
 	return os.Stat(fname)
 }
 
+func (realOS) Statfs(fname string) (syscall.Statfs_t, error) {
+	var stat syscall.Statfs_t
+	err := syscall.Statfs(fname, &stat)
```

Comment:
It looks like `Statfs()` and `Statfs_t` are not portable to all systems. Will have to find something that works across the board.

Reply:
Perhaps this library is a good candidate: https://github.com/shirou/gopsutil

Reply:
that dependency is quite large, can we use also syscall on Windows (a different one):

https://stackoverflow.com/questions/20108520/get-amount-of-free-disk-space-using-go

This requires conditional compilation - you can put Statfs() in separate files, one for Windows, and one for non-Windows.

Reply:
It looks like we'll have to define this call at least thrice: `windows`, `openbsd_arm64`, and `linux_arm64`. At the risk of writing a lot of code that I'd like not to be in our purview to maintain. What does everyone think of importing just `GitHub.com/shirou/gopsutil/disk`? That sub package contains just the definitions we need (albeit for a larger set of architectures than we ourselves test/support).